protocol/communities/community.go

```diff
@@ -1437,9 +1441,11 @@ func (o *Community) AddTokenPermission(permission *protobuf.CommunityTokenPermis
 	defer o.mutex.Unlock()
 
 	isControlNode := o.IsControlNode()
-	allowedToSendEvents := o.HasPermissionToSendCommunityEvents()
+	allowedToManageMemberPermissions := o.IsTokenMaster() || o.IsAdmin()
+	allowedToManageAllPermissions := o.IsOwnerWithoutCommunityKey()
 
-	if !isControlNode && !allowedToSendEvents || (allowedToSendEvents && permission.Type == protobuf.CommunityTokenPermission_BECOME_ADMIN) {
+	if !isControlNode && !allowedToManageMemberPermissions && !allowedToManageAllPermissions ||
```

Comment:
It just makes the logic easier to follow
```suggestion
	if (!isControlNode && !allowedToManageMemberPermissions && !allowedToManageAllPermissions) ||
```